hs_dataverse/utils.py

```diff
@@ -0,0 +1,672 @@
+        geo_coverage_vals.append(geo_coverage_dict)
+
+    # Extract more fields using the extended metadata from other_metadata.json
+    e = dict()
+    with open('/'.join([temp_dir, 'other_metadata.json'])) as f:
+        e = json.load(f)
+
+    other_id_dict['otherIdValue']['value'] = str(e['rid'])
+
+    notes_text = e['extended_metadata_notes']
+
+    grant_vals = []
+    for i, grant in enumerate(e['award_numbers']):
+        grant_info = copy.deepcopy(grant_number_dict)
+        grant_info['grantNumberAgency']['value'] = e['funding_agency_names'][i]
+        grant_info['grantNumberValue']['value'] = e['award_numbers'][i]
+        grant_vals.append(grant_info)
+
+    contributor_vals = []
+    for contributor in e['contributors']:
+        c_dict = copy.deepcopy(contributor_dict)
+        c_dict['contributorName']['value'] = contributor
+        contributor_vals.append(c_dict)
+
+    # Extract more fields using the owner data from ownerdata.json
+    o = dict()
+    with open('/'.join([temp_dir, 'ownerdata.json'])) as f:
+        o = json.load(f)
+    contact = copy.deepcopy(contact_dict)
+    if 'username' in o:
+        producer_dict['producerAbbreviation']['value'] = o['username']
+    else:
+        producer_dict['producerAbbreviation']['value'] = ''
+
+    if 'first_name' in o and 'last_name' in o:
+        producer_dict['producerName']['value'] = o['last_name'] + ', ' + o['first_name']
+        contact['datasetContactName']['value'] = o['last_name'] + ', ' + o['first_name']
+        depositor = o['last_name'] + ', ' + o['first_name']
+    else:
+        producer_dict['producerName']['value'] = ''
+        contact['datasetContactName']['value'] = ''
+        depositor = ''
+
+    if 'organization' in o:
+        producer_dict['producerAffiliation']['value'] = o['organization']
+        contact['datasetContactAffiliation']['value'] = o['organization']
+    else:
+        producer_dict['producerAffiliation']['value'] = ''
+        contact['datasetContactAffiliation']['value'] = ''
+
+    if 'email' in o:
+        contact['datasetContactEmail']['value'] = o['email']
+    else:
+        contact['datasetContactEmail']['value'] = ''
+    if contact != contact_dict:
+        contact_vals.append(contact)
+
+    # update the json dict with the field values
+    fields[0]['value'] = title
+
+    fields[1]['value'] = alt_url
+
+    fields[2]['value'] = [other_id_dict]
+
+    fields[3]['value'] = author_vals
+    fields[4]['value'] = contact_vals
+
+    fields[5]['value'][0]['dsDescriptionValue']['value'] = abstract
+    fields[5]['value'][0]['dsDescriptionDate']['value'] = last_modified_date
+
+    fields[6]['value'] = subject
+
+    fields[7]['value'] = keyword_vals
+
+    fields[8]['value'] = related_publications_vals
+
+    fields[9]['value'] = notes_text
+
+    fields[10]['value'] = [producer_dict]
+
+    fields[11]['value'] = contributor_vals
+
+    fields[12]['value'] = grant_vals
+
+    fields[13]['value'] = [distributor_dict]
```

Comment:
This is super brittle. It depends upon prior knowledge of structure. Either tag lookup or Django template this.